### PR TITLE
Refactor the documentation

### PR DESCRIPTION
**Pull request by josh146**
**Context:** Refactors the documentation to take into account the new quantum photonics page.

**Description of the Change:**

* Algorithms section is deleted

* Gallery is deleted

* Tutorials are deleted

* Photonic theory is deleted

* Intersphinx is set up, so that cross-references can be made to the photonics repo

* Sphinx-Gallery is removed from the requirements

* TensorFlow is unmocked out, which will help with documentation building.

* Navbar has been updated.

**Benefits:** n/a

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

**Comment by josh146** [comment]
Note: the rendered documentation will be available at https://strawberryfields.readthedocs.io/en/prune-documentation once RTDs gets its act together.

File: doc/index.rst [hunk 1]
```diff
@@ -149,15 +149,10 @@ Strawberry Fields is **free** and **open source**, released under the Apache Lic
...
-   introduction/graphs
-   introduction/ml
-   introduction/chemistry
-   introduction/sampling
    introduction/circuits
    introduction/ops
    introduction/states
-   introduction/tutorials
-   zreferences
+   introduction/data
```
Comment (trbromley) [inline doc/index.rst:155]: Thanks for remembering GBS datasets!

File: doc/introduction/introduction.rst [hunk 2]
```diff
@@ -1,162 +1,71 @@
...
-    .sphx-glr-thumbcontainer .figure {
-      width: 200px;
-    }
-    .sphx-glr-thumbcontainer img {
-      width: 100%;
-    }
-    </style>
-
 .. image:: /_static/NOON.png
     :align: left
```
Comment (trbromley) [inline doc/introduction/introduction.rst:5]:
```suggestion
    :align: right
```
This it might look better right-aligned?
image

File: doc/introduction/introduction.rst [hunk 3]
```diff
@@ -1,162 +1,71 @@
...
+   when using the built-in simulators. These provide useful information and processing
+   of the quantum state of the system.
 
-.. raw:: html
+..
 
-    <a href="intro_photonics/op_conventions.html">
-    <h5>read more <i class="fas fa-angle-double-right"></i></h5>
-    </a>
+5. :doc:`photonic_hardware` gives an overview of building, submitting, and managing
```
Comment (trbromley) [inline doc/introduction/introduction.rst:64]: This could be the first one, since that's how it appears in the table of contents

File: doc/xanadu_theme/header.html [hunk 4]
```diff
@@ -41,18 +44,18 @@
 
     <ul class="navbar-nav ml-auto nav-flex-icons">
       <li class="nav-item">
-        <a class="nav-link" href="https://u.strawberryfields.ai/slack">
-          <i class="fab fa-slack"></i> Slack
+        <a class="nav-link" href="/faq.html">
+          <i class="fab fas fa-question pr-1"></i> FAQ
```
Comment (trbromley) [inline doc/xanadu_theme/header.html:48]: :thinking: this is new?

Comment (trbromley) [reply doc/xanadu_theme/header.html:48]: https://pennylane.ai/faq.html the objective is a similar thing to this?

File: examples_apps/run_tutorial_similarity.py [hunk 5]
```diff
@@ -1,314 +0,0 @@
-# pylint: disable=wrong-import-position,wrong-import-order,ungrouped-imports,invalid-name
```
Comment (trbromley) [inline examples_apps/run_tutorial_similarity.py:1]: One thing to bear in mind is that https://github.com/XanaduAI/strawberryfields/pull/390 wants to change this tutorial :thinking: I guess we can just do that in a separate PR to photonic-applications?

File: doc/conf.py [hunk 6]
```diff
@@ -54,14 +43,15 @@ def __getattr__(cls, name):
...
-    'nbsphinx',
-    'sphinx_gallery.gen_gallery',
     "sphinx.ext.intersphinx",
     "sphinx_automodapi.automodapi",
     'sphinx_copybutton',
     "m2r"
 ]
 
+
+intersphinx_mapping = {"photonics": ("http://sfdocsredesign.s3-website-us-east-1.amazonaws.com/photonics", None)}
```
[the remark this answers is not in the document]
Comment (josh146) [reply doc/conf.py:53]: Was making the assumption that no one will go looking in the deep dark depths of this PRs changed files, so I might delete this comment :) This was required so that intersphinx would work properly, but this will be changed to point to sf.ai before this is merged.

File: doc/introduction/introduction.rst (hunk 3 above)
Comment (josh146) [reply doc/introduction/introduction.rst:64]: Oh, I was thinking of moving the hardware quickstart to 5th in the TOC, partly because I was looking at it, and realised it had assumed knowledge on circuits and operations etc. So it makes a bit more sense for the other quickstarts to come first.

File: examples_apps/run_tutorial_similarity.py (hunk 5 above)
Comment (josh146) [reply examples_apps/run_tutorial_similarity.py:1]: Yep!
